project/frontend/src/FindMatch/FindMatch.js

```diff
@@ -129,4 +149,124 @@ class PreferenceForm extends React.Component {
       </form>
     );
   }
+
+  getRecommendation() {
+    const promises = this.makePromisesArray();
+    const preferences = this.makePrefsBody();
+
+    Promise.all(promises)
+      // This gives us the list of restaurants.
+      .then((responses) =>
+        Promise.all(responses.map((response) => response.json()))
+      )
+      .then((data) => {
+        let restaurants = [];
+        for (const restaurant of data) {
+          restaurants = restaurants.concat(restaurant.results);
+        }
+
+        fetch('/api/recommendation', {
+          method: 'POST',
+          headers: {
+            'Access-Control-Allow-Origin': '*',
+            'Content-Type': 'application/json',
+          },
+          body: JSON.stringify({
+            restaurants,
+            preferences,
+          }),
+        })
+          .then((response) => response.text())
+          .then((data) => {
+            try {
+              const selections = JSON.parse(data);
+              console.log(selections);
+            } catch (err) {
+              console.log(err);
+            }
+          });
+      });
+  }
+
+  /**
+   * Creates and returns an object to be used in the request body sent to the servlet.
+   */
+  makePrefsBody() {
+    const lat = parseFloat(this.state.latitude);
```

Comment:
nit: pull these variables out so we don't have to write this.state... throughout this function. `const { latitude, longitude, distance, ... etc } = this.state`